Formula/xorg-server.rb

```diff
@@ -0,0 +1,107 @@
+class XorgServer < Formula
+  desc "X Window System display server"
+  homepage "https://www.x.org"
+  url "https://www.x.org/releases/individual/xserver/xorg-server-21.1.4.tar.xz"
+  sha256 "5cc4be8ee47edb58d4a90e603a59d56b40291ad38371b0bd2471fc3cbee1c587"
+  license all_of: ["MIT", "APSL-2.0"]
+
+  depends_on "font-util"   => :build
+  depends_on "libxkbfile"  => :build
+  depends_on "meson"       => :build
+  depends_on "ninja"       => :build
+  depends_on "pkg-config"  => :build
+  depends_on "util-macros" => :build
+  depends_on "xorgproto"   => :build
+  depends_on "xtrans"      => :build
+
+  depends_on "libxfixes"
+  depends_on "libxfont2"
+  depends_on "mesa"
+  depends_on "pixman"
+  depends_on "xcb-util"
+  depends_on "xcb-util-image"
+  depends_on "xcb-util-keysyms"
+  depends_on "xcb-util-renderutil"
+  depends_on "xcb-util-wm"
+  depends_on "xkbcomp"
+  depends_on "xkeyboardconfig"
+
+  on_macos do
+    depends_on "libapplewm"
+  end
+
+  on_linux do
+    depends_on "dbus"
+    depends_on "libdrm"
+    depends_on "libepoxy"
+    depends_on "libxcvt"
+    depends_on "libxshmfence"
+    depends_on "nettle"
+    depends_on "systemd"
+  end
+
+  def install
+    # ChangeLog contains some non relocatable strings
+    rm "ChangeLog"
+    # set proper font dir
+    inreplace "meson.build", "fontutil_dep.get_pkgconfig_variable('fontrootdir')",
+              "'#{HOMEBREW_PREFIX}/share/fonts/X11'"
+    # avoid setting `xkb_dir`
+    ENV.remove "PKG_CONFIG_PATH", Formula["xkbcomp"].opt_lib/"pkgconfig"
+    inreplace "meson.build", "dependency('xkbcomp', required: false)",
+              "dependency('xxxxx', required: false)"
```

Comment:
`ENV.remove "PKG_CONFIG_PATH", Formula["xkbcomp"].opt_lib/"pkgconfig"` doesn't work on Linux.

Reply:
Can you detail what you are trying to do here? What is `xkb_dir` and why are you avoiding it?

Reply:
From [`meson.build`](https://gitlab.freedesktop.org/xorg/xserver/-/blob/master/meson.build#L124)
```meson
xkb_dir = get_option('xkb_dir')
if xkb_dir == ''
    if xkbcomp_dep.found() and xkbcomp_dep.type_name() == 'pkgconfig'
        xkb_dir = xkbcomp_dep.get_pkgconfig_variable('xkbconfigdir')
    endif
    if xkb_dir == ''
        xkb_dir = join_paths(get_option('prefix'), 'share/X11/xkb')
    endif
endif
```
It trys to use `xkbcomp` to deduce the xkb related files location, e.g. files from `xkeyboardconfig`, `xkbcomp` paths from `pkg-config` contains `Cellar` so it cannot find files from `xkeyboardconfig`.

Reply:
Potentially another candidate for adjusting the pkg-config on the `xkbcomp` side (assuming it doesn't break anything else).

Reply:
> Potentially another candidate for adjusting the pkg-config on the `xkbcomp` side (assuming it doesn't break anything else).

Fixed in #109228.